### DISCUSSION TITLE
Can I put ms dos 2.0 my source code in my reprository?

### DISCUSSION THREAD

**Discussion by kurlyak** [community/community#58274] [Repositories] [answered]
### Select Topic Area

Question

### Body

I have question to administration.

I did a disassembly of the ms dos 2.0 boot sector, this is the boot.asm file for the NASM assembler compiler. Do I have the right to put this boot.asm file and the IBMBIO.COM, IBMDOS.COM, COMMAND.COM files in my GitHub repository, or does this violate the license agreement?

And second question. Can I put ms dos 2.11 boot sector source code, and IO.SYS, MSDOS.SYS, COMMAND.COM to my reprository?

https://github.com/Microsoft/MS-DOS/blob/master/LICENSE.md

**Comment by the-officialjosh** [comment, marked as answer]
According to the license file available in the MS-DOS v1.25 and v2.0 Source Code repository, these files are released under the MIT License. The MIT License grants the right to use, copy, modify, merge, publish, distribute, sublicense, and sell the software.

As per the MIT License, there are no specific restrictions mentioned regarding the inclusion of the `boot.asm` file, `IBMBIO.COM`, `IBMDOS.COM`, `COMMAND.COM` files (for MS-DOS 2.0), or the source code for MS-DOS 2.11 in your GitHub repository.

However, it's always recommended to carefully review the license in its entirety and ensure compliance with all its conditions. If you have any specific concerns or questions about the licensing and usage, it's advisable to consult legal experts or seek further guidance to ensure full compliance with the license and applicable laws.

Please note that this response is based on the information provided in the license file, and it's important to thoroughly review the license and consider seeking legal advice if needed.

**Comment by kurlyak** [comment]
Thank you very much the-officialjosh, more likely I will publish MS DOS 2.11 on my reprository. With my wource code MS DOS 2.11, I will publish imdisk.exe and nasm.exe in same reprository (so that the user can easily make their own IMG image for VirtualBox). nasm.exe is known software, and imdisk.exe is link:

https://sourceforge.net/projects/imdisk-toolkit/

Can I place nasm.exe and imdisk.exe into my reprository with source code? It's not violate GitHub rules?

**Reply by the-officialjosh** [reply]
It is an open-sourced project so no. Just read through their license carefully that's all. You should be good to go. You can build on an existing open-source as you like. Please mark the earlier response as the answer for those who need help with similar issues to easily find this

**Comment by kurlyak** [comment]
Thank you very much for answer.

I provided screenshot when loading my IMG MS DOS 2.11:

https://ibb.co/w6vSSsp

it is written there as usual - Microsoft MS DOS version 2.11 and then Default Device Driver Revision 1.0 (c) Copyright Data General Corporation 1984 all rights reserved. Can I ask again more detailed? Can I put IO.SYS and MSDOS.SYS, COMMAND.COM and my boot.asm  code to my GitHub reprository? Written Data General Corporaton all right reserved. Or only my boot.asm file without MS DOS system files?

**Reply by the-officialjosh** [reply]
What I was advising was to be careful when proceeding, but considering it is public on GitHub, it is most likely safe for personal use. If there are other licenses attached, most of the time the main reason is just against commercial use and redistribution, not personal use. You can read through their other licenses anyway just to be sure. Good luck! 👌

**Comment by kurlyak** [comment]
Thank you very much, the-officialjosh. I a bit doubt.

https://github.com/Microsoft/MS-DOS/blob/master/LICENSE.md

There is licence agreement and it says:

"Permission is hereby granted, free of charge, to any person obtaining a copy of this software"

It meand ANY copy or only THIS copy? It says "a copy of this software" i.e. sofware placed on GitHub Microsoft? Or any copy MS DOS 2.0 taken from Internet? I want place origianl IMG file with bootable MS DOS 2.0 to my GitHub Reprository. Can I do it?

The license refers to any copy MS DOS 2.0 taken from Internet?

UPD.
Is this license only for the code that Microsoft has on GitHub? I.e. here is this source code (and only it no other copies of MS DOS 2.0)

https://github.com/microsoft/MS-DOS

And here is the license for this source code (and this licence dont supports MS DOS 2.0 Olivetti for example).

https://github.com/Microsoft/MS-DOS/blob/master/LICENSE.md

You see, I will contact a lawyer here at my place of residence, and then I will be banned from GitHub on the Internet. Here you need a lawyer from Microsoft or GitHub.

For example, the MS DOS 2.0 Olivetti includes files that were made by IBM - these are the files IBMBIO.COM, IBMDOS.COM - how can Microsoft give the permission to use IBM files? Different OEMs made different MS DOS 2.0. Does the license cover all OEM builds?
